src/Security/src/DataProtection.Redis/RedisDataProtectionBuilderExtensions.cs

```diff
@@ -26,19 +31,61 @@ public static class RedisDataProtectionBuilderExtensions
     /// The incoming <paramref name="builder" /> so that additional calls can be chained.
     /// </returns>
     public static IDataProtectionBuilder PersistKeysToRedis(this IDataProtectionBuilder builder)
+    {
+        return PersistKeysToRedis(builder, string.Empty);
+    }
+
+    /// <summary>
+    /// Configures the data protection system to persist keys in a Redis database, using the Steeltoe Connector for Redis.
+    /// </summary>
+    /// <param name="builder">
+    /// The <see cref="IDataProtectionBuilder" /> to configure.
+    /// </param>
+    /// <param name="serviceBindingName">
+    /// The service binding name, in case multiple redis services are used.
+    /// </param>
+    /// <returns>
+    /// The incoming <paramref name="builder" /> so that additional calls can be chained.
+    /// </returns>
+    public static IDataProtectionBuilder PersistKeysToRedis(this IDataProtectionBuilder builder, string serviceBindingName)
     {
         ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(serviceBindingName);
 
-        builder.Services.TryAddSingleton(provider =>
-        {
-            var connectorFactory = provider.GetRequiredService<ConnectorFactory<RedisOptions, IDistributedCache>>();
-            Connector<RedisOptions, IDistributedCache> connector = connectorFactory.Get();
-            return connector.GetConnection();
-        });
+        builder.Services.AddStackExchangeRedisCache(EmptyAction);
```

Comment:
Is there any potential conflict with `Services.AddStackExchangeRedisCache` being called separately in an application?

Reply:
Shouldn't hurt. It just registers an IOptions callback, so when done multiple times, multiple callback will run.

Reply:
This is what's behind it:
```c#
services.AddOptions();
services.Configure(setupAction);
services.Add(ServiceDescriptor.Singleton<IDistributedCache, RedisCacheImpl>());
```